src/mudlet.cpp

```diff
@@ -1539,11 +1535,6 @@ void mudlet::slot_close_profile_requested(int tab)
 
     migrateDebugConsole(pH);
 
-    // Wait for disconnection to complete
-    while (pH->mTelnet.getConnectionState() != QAbstractSocket::UnconnectedState) {
```

Comment:
:rotating_light: Hang on - we put these waits in so that Mudlet wouldn't hang when disconnection from a secure connection or something like that as #2376 - so won't we get **that** problem back again?

Reply:
This is exactly what was causing this crash / hang.

We won't get that problem back again because I applied the solution that was Windows-only - the aborts - to all platforms.

Reply:
Feel free to test, though! It works well for me, but more practical testing would always help.